openmicroscopy/openmicroscopy
# Short url

Original work started in #4401, this replace #4420

-----

This PR is a proposal for https://trello.com/c/sJCSdD7P/174-redirect-service

To test:

- create short url for ``/webclient/?show=image-1`` (stored as omero.model.Alias)

 ```
c=omero.client('localhost', 4064)
sess=c.createSession('root', 'secret')
conn=omero.gateway.BlitzGateway(client_obj=c)
update = conn.getUpdateService()
ns = "host/port/prefix" #or referrer
short = omero.model.AliasI()
short.setNs(omero.rtypes.rstring(ns))
short.origin = omero.rtypes.rstring("/webclient/?show=image-1")
short.details.permissions = omero.model.PermissionsI("rwr-r-")
short = update.saveAndReturnObject(short)
print short.id.val, short.origin.val
print 'short', base62.from_decimal(short.id.val)
 ```

- go to https://host:port/prefix/url/_SHORT_
- check if you were redirected to the expected location

Example: Alias:ID = 1
```
$ curl -I http://localhost:4080/url/B
HTTP/1.0 302 FOUND
Date: Wed, 20 Jan 2016 13:40:14 GMT
Server: WSGIServer/0.1 Python/2.7.10
Vary: Cookie
Content-Type: text/html; charset=utf-8
Location: http://localhost:4080/webclient/?show=image-1
```

Future work:
- storing *referrer*
- custom *name*
- UI
